File: ios/Classes/BannerView/BannerViewFactory.swift
//
//  BannerViewFactory.swift
//  clever_ads_solutions
//
//  Copyright © 2024 CleverAdsSolutions LTD, CAS.AI. All rights reserved.
//

import Flutter
import Foundation

class BannerViewFactory: NSObject, FlutterPlatformViewFactory {
    private let registrar: FlutterPluginRegistrar
    private let managerHandler: MediationManagerMethodHandler
    private let methodHandler: BannerMethodHandler

    init(with registrar: FlutterPluginRegistrar, managerHandler: MediationManagerMethodHandler) {
        self.registrar = registrar
        self.managerHandler = managerHandler
        methodHandler = BannerMethodHandler(with: registrar)
    }

    func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        return BannerView(
            frame: frame,
            viewId: viewId,
            args: args as? [String: Any?],
            manager: managerHandler.manager,
            methodHandler: methodHandler
        )
    }

    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}

File: android/src/main/kotlin/com/cleveradssolutions/plugin/flutter/bannerview/BannerViewFactory.kt
package com.cleveradssolutions.plugin.flutter.bannerview

import android.content.Context
import com.cleveradssolutions.plugin.flutter.bridge.manager.MediationManagerMethodHandler
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class BannerViewFactory(
    binding: FlutterPluginBinding,
    private val managerHandler: MediationManagerMethodHandler
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    private val methodHandler = BannerMethodHandler(binding)

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        return BannerView(
            context,
            viewId,
            args as Map<*, *>?,
            managerHandler.manager,
            methodHandler
        )
    }

}